tests/driver_epd_bw_spi/Makefile

```diff
@@ -0,0 +1,16 @@
+include ../Makefile.tests_common
+
+BOARD := nucleo-f411re
+
+USEMODULE += xtimer
+USEMODULE += epd_bw_spi
+
+CFLAGS += -DDISPLAY_X=200
+CFLAGS += -DDISPLAY_Y=200
+CFLAGS += -DEPD_BW_SPI_CONTROLLER=EPD_BW_SPI_CONTROLLER_IL3829
+CFLAGS += -DEPD_BW_SPI_PARAM_CS=GPIO_PIN\(PORT_B,6\)
+CFLAGS += -DEPD_BW_SPI_PARAM_DC=GPIO_PIN\(PORT_C,7\)
+CFLAGS += -DEPD_BW_SPI_PARAM_RST=GPIO_PIN\(PORT_A,8\)
+CFLAGS += -DEPD_BW_SPI_PARAM_BUSY=GPIO_PIN\(PORT_B,10\)
```

Comment:
I would not hardcode all these parameters in a generic application Makefile. I have an STM32 board with a compatible screen (and a branch ;) ) where I would like to define all these params in the board configuration. The problem with the approach here is that it will conflict.

Reply:
Do you have a suggestion about where to define it for people (like me) that don't have a board with a display?

Reply:
On projects that require custom config for a board I usually do create a `board.h` fine and `#include_next "board.h"` on it. E.g:

- board.h
```c
#ifdef BOARD_BOARD_NUCLEO_F411RE
#define EPD_BW_SPI_CONTROLLER=EPD_BW_SPI_CONTROLLER_IL3829
#define EPD_BW_SPI_PARAM_CS=GPIO_PIN(PORT_B,6)
#define EPD_BW_SPI_PARAM_DC=GPIO_PIN(PORT_C,7)
#define EPD_BW_SPI_PARAM_RST=GPIO_PIN(PORT_A,8)
#define EPD_BW_SPI_PARAM_BUSY=GPIO_PIN(PORT_B,10)
#endif

#include_next "board.h"
```

And add to applications Makefile `INCLUDES += -I$(APPDIR)`.

Reply:
See also [this](https://doc.riot-os.org/group__cpu__esp8266.html#esp8266_application_specific_board_configuration) and [this](https://doc.riot-os.org/group__cpu__esp8266.html#esp8266_application_specific_driver_configuration) which applies to all boards I think.

Reply:
The usual way in RIOT for this kind of driver parameters is to define them (with default values) in `<driver>_params.h`. Then, if a board requires specific parameters, they can be defined in the board.h header of the board.

See the [driver implementation guidelines](http://doc.riot-os.org/driver-guide.html), see the `Default device configuration` paragraph where this is explained.